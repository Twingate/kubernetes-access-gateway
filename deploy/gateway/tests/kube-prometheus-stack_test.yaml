suite: Kube Prometheus Stack
templates:
  - charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml
tests:
  - it: should enable Kube Prometheus Stack
    # Hardcode the chart version and app version to ensure that the snapshot is stable
    # when upgrading Kube Prometheus Stack subchart
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    set:
      kube-prometheus-stack:
        enabled: true
    asserts:
      - matchSnapshot: {}
