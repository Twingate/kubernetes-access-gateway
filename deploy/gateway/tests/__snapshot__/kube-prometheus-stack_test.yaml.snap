should enable Kube Prometheus Stack:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: Prometheus
    metadata:
      labels:
        app: kube-prometheus-stack-prometheus
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: kube-prometheus-stack
        app.kubernetes.io/version: 1.0.0
        chart: kube-prometheus-stack-1.0.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-kube-promethe-prometheus
      namespace: NAMESPACE
    spec:
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
            - podAffinityTerm:
                labelSelector:
                  matchExpressions:
                    - key: app.kubernetes.io/name
                      operator: In
                      values:
                        - prometheus
                    - key: app.kubernetes.io/instance
                      operator: In
                      values:
                        - RELEASE-NAME-kube-promethe-prometheus
                topologyKey: kubernetes.io/hostname
              weight: 100
      alerting:
        alertmanagers:
          - apiVersion: v2
            name: RELEASE-NAME-kube-promethe-alertmanager
            namespace: NAMESPACE
            pathPrefix: /
            port: http-web
      automountServiceAccountToken: true
      enableAdminAPI: false
      enableOTLPReceiver: false
      externalUrl: http://RELEASE-NAME-kube-promethe-prometheus.NAMESPACE:9090
      hostNetwork: false
      image: quay.io/prometheus/prometheus:v3.5.0
      imagePullPolicy: IfNotPresent
      listenLocal: false
      logFormat: logfmt
      logLevel: info
      paused: false
      podMonitorNamespaceSelector: {}
      podMonitorSelector: {}
      portName: http-web
      probeNamespaceSelector: {}
      probeSelector:
        matchLabels:
          release: RELEASE-NAME
      replicas: 1
      retention: 10d
      routePrefix: /
      ruleNamespaceSelector: {}
      ruleSelector: {}
      scrapeConfigNamespaceSelector: {}
      scrapeConfigSelector:
        matchLabels:
          release: RELEASE-NAME
      securityContext:
        fsGroup: 2000
        runAsGroup: 2000
        runAsNonRoot: true
        runAsUser: 1000
        seccompProfile:
          type: RuntimeDefault
      serviceAccountName: RELEASE-NAME-kube-promethe-prometheus
      serviceMonitorNamespaceSelector: {}
      serviceMonitorSelector: {}
      shards: 1
      tsdb:
        outOfOrderTimeWindow: 0s
      version: v3.5.0
      walCompression: true
